### PR TITLE
[loganalyzer] [Qos] Ignore brcm_sai_set_switch_attribute error in syslog

### PR DESCRIPTION
**Pull request by wenyiz2021**
### Description of PR

Ignore brcm_sai error in syslog for QOS tests, when disabling loganalyzer, tests are passing normally, with loganalyzer there is error which does not impact test behavior:
2022-05-10T21:01:39.1386482Z E               LogAnalyzerError: {'match_messages': {'/tmp/syslog.masic.2022-05-10-19:12:50': ['May 10 19:12:16.410561 masic ERR syncd0#syncd: [03:00.0] brcm_sai_set_switch_attribute:1920 updating switch mac addr failed with error -2.\n']}, 'total': {'expected_match': 0, 'expected_missing_match': 0, 'match': 1}, 'match_files': {'/tmp/syslog.masic.2022-05-10-19:12:50': {'expected_match': 0, 'match': 1}}, 'expect_messages': {'/tmp/syslog.masic.2022-05-10-19:12:50': []}, 'unused_expected_regexp': []}

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [X] 202012

### CODE REVIEW

**Comment by wenyiz2021** [comment]
Test result before:
ERROR qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[0-xoff_1] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost svcstr-n3164-acs-1>']" failed with exit code "1"
ERROR qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[0-xoff_2] - Failed: Processes "['analyze_logs--<MultiAsicSonicHost svcstr-n3164-acs-1>']" failed with exit code "1"
After:
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[0-xoff_1] PASSED                                                                                                                                                                           [  2%]
qos/test_qos_sai.py::TestQosSai::testQosSaiPfcXoffLimit[0-xoff_2] PASSED                                                                                                                                                                           [  3%]

File: tests/qos/test_qos_sai.py [hunk 1]
```diff
@@ -38,6 +38,15 @@
 
 PTF_PORT_MAPPING_MODE = 'use_orig_interface'
 
+@pytest.fixture(autouse=True)
+def ignore_expected_loganalyzer_exception(rand_one_dut_hostname, loganalyzer):
+    """ignore the syslog ERR syncd0#syncd: [03:00.0] brcm_sai_set_switch_attribute:1920 updating switch mac addr failed with error -2"""
+    ignore_regex = [
+            ".*ERR syncd[0-9]*#syncd.*brcm_sai*"
```
Comment (neethajohn) [inline tests/qos/test_qos_sai.py:45]: please make this msg specific to ignore only the switch mac addr failed.

Comment (wenyiz2021) [reply tests/qos/test_qos_sai.py]: > please make this msg specific to ignore only the switch mac addr failed.
thanks for pointing out, done

File: tests/qos/test_qos_sai.py [hunk 2]
```diff
@@ -38,6 +38,15 @@
 
 PTF_PORT_MAPPING_MODE = 'use_orig_interface'
 
+@pytest.fixture(autouse=True)
+def ignore_expected_loganalyzer_exception(rand_one_dut_hostname, loganalyzer):
+    """ignore the syslog ERR syncd0#syncd: [03:00.0] brcm_sai_set_switch_attribute:1920 updating switch mac addr failed with error -2"""
+    ignore_regex = [
+            ".*ERR syncd[0-9]*#syncd.*brcm_sai_set_switch_attribute*"
```
Comment (neethajohn) [inline tests/qos/test_qos_sai.py:45]: please also include "updating switch mac addr failed"  in your regex. we only want to ignore that particular error

Comment (wenyiz2021) [reply tests/qos/test_qos_sai.py]: Thanks Nejo, done.

**Review by neethajohn — APPROVED** [review]

**Pull request merged**